lib/dgram.js

```diff
@@ -78,7 +78,7 @@ exports._createSocketHandle = function(address, port, addressType, fd, flags) {
 function Socket(type, listener) {
   EventEmitter.call(this);
 
-  if (typeof type === 'object') {
+  if (type && typeof type === 'object') {
```

Comment:
@cjihrig do you mean `type && type !== null && type === 'object'` ? Or in some other way?